content/en/docs/contributing/policy.md

```diff
@@ -0,0 +1,61 @@
+
+# Features we will not allow
+
+Certain features have ben requested before but are not able to being implemented properly or not able to being implemented without breaking the security model.
+
+Below is a list of a few of these:
+
+## Vendoring Kubernetes related APIs outside of the `k8s.io/` namespace
+
+Vendoring Kubernetes related APIs like OpenShift specific classes is not recommend as it vendors the `k8s.io/apimachinery` package itself could potentially conflict with cert-manager's instance. It could also cause a conflict with different Kubernetes client versions being used.
+
+If this is needed it is suggested to use a "dynamic client" that converts the objects into internal structured copied into the cert-manager codebase.
+
+## Helm + CRDs
+
+Currently Helm suggests CRDs is to include them in the `crds/` subdirectory of a chart with the `crd-install` annotation included.
+However this has the side effect that CRDs will not be upgraded any more if changes are made in a later release.
+CRDs being upgraded without them being removed and re-installed is essential for cert-manager to move forward.
+The mechanics of this are currently being discussed [in the Helm community](https://github.com/helm/helm/issues/5871).
+
+cert-manager currently works around this by shipping the CRDs in the templates. 
+
+## Helm Subchart capabilities
+
+cert-manager should not be used as a sub-chart in Helm deployments.
+Helm deployments are namespaced by design and do not support a dependency model that would support cert-manager being operated on a cluster scope nor preventing cert-manager from being installed twice (with conflicting versions) or being upgraded independently from the application.
+
+The cert-manager installation creates cluster scoped resources like admission webhooks and custom resource definitions. cert-manager should be seen as part of your cluster and should be treated as such for being installed. A comparison to other Kubernetes components would be a LoadBalancer controller or a PV provisioner
+
+## Secret injection or copying
+
+cert-manager deals with very sensitive information (all TLS certificates for your services) and has cluster-level access to secret resources, therefore when designing features we need to consider all ways these can be abused to escalate privilege.
+Secret data is meant to be securely stored in the secret resources and have narrow scoped access privileges for unauthorized users. Therefore we will not allow any functionality that allows this data to be copied/injected into any non secret resource. 
+
+### cainjector
+
+The cainjector component is a special exception to this rule. This component is able to inject the `ca.crt` file into predefined fields on `ValidatingWebhookConfiguration`, `MutatingWebhookConfiguration`, and `CustomResourceDefinition` resources from Certificate resources.
```

Comment:
If you're open to injecting into configmaps (I think this is probably reasonable), I would indicate that configmaps are anticipated to be the outward boundary/API instead of injecting into resources managed by other projects.

Reply:
Going to leave that out for now as there is a discussion going on about configmaps